.github/workflows/ci.yml

```diff
@@ -73,3 +73,52 @@ jobs:
       - name: Run tests
         run: |
           python -m pytest --cov=marray
+
+  array_api_tests:
+    needs: detect-ci-trigger
+    runs-on: ${{ matrix.os }}
+    if: needs.detect-ci-trigger.outputs.triggered == 'false'
+
+    strategy:
+      fail-fast: false
+
+      matrix:
+        os: ["ubuntu-latest", "macos-latest", "windows-latest"]
+        python-version: ["3.10", "3.11", "3.12"]
+
+    steps:
+      - name: Clone repository
+        uses: actions/checkout@v4
+
+      - name: Checkout array-api-tests
+        uses: actions/checkout@v4
+        with:
+          repository: data-apis/array-api-tests
+          ref: "8d4ccd63d860ba7202cb5ae1282aaf37d5578b10" # Latest commit as of 2024-12-27
+          submodules: "true"
+          path: "array-api-tests"
+          persist-credentials: false
+
+      - name: Setup environment
+        uses: actions/setup-python@v5
+        with:
+          python-version: "${{ matrix.python-version }}"
+          cache: "pip"
+
+      - name: Install dependencies
+        run: |
+          python -m pip install -r ci/requirements/environment.txt
+          python -m pip install -r array-api-tests/requirements.txt
+
+      - name: Install marray
+        run: |
+          python -m pip install --no-deps -e .
+
+      - name: Setup test suite
+        run: |
+          cd ${GITHUB_WORKSPACE}/array-api-tests
+          git apply ${GITHUB_WORKSPACE}/ci/xp-tests.patch
```

Comment:
This is a bit hacky. I wonder whether I could create a GitHub Action to automate the patching for a given array-api-tests version.

Reply:
That would be very cool! 
I'll make you a maintainer so CI runs for you. 
Will there be a way to skip tests? There will probably be a few that are hard to get working (e.g. if there are some with dlpack).

Reply:
yep, https://github.com/numpy/numpy/blob/main/tools/ci/array-api-xfails.txt

Reply:
OK, ran CI and made you a collaborator so you can play around with this.

Reply:
thanks!